airbyte-webapp/src/hooks/useAnalytics.tsx

```diff
@@ -1,25 +1,64 @@
-import React, { useContext, useMemo } from "react";
+import React, { useContext, useMemo, useState } from "react";
 import { AnalyticsService } from "core/analytics/AnalyticsService";
 
-const analyticsServiceContext = React.createContext<AnalyticsService | null>(
+export type AnalyticsServiceProviderValue = {
+  setContext: (ctx: Record<string, unknown>) => void;
+  addContextProps: (props: Record<string, unknown>) => void;
+  removeContextProps: (props: string[]) => void;
+  service: AnalyticsService;
+};
+
+const analyticsServiceContext = React.createContext<AnalyticsServiceProviderValue | null>(
   null
 );
 
 function AnalyticsServiceProvider({
   children,
-  userId,
   version,
+  context,
 }: {
   children: React.ReactNode;
   version?: string;
-  userId?: string;
+  context?: Record<string, unknown>;
```

Comment:
we don't want to set all context at once. This prop makes it a bit confusing that you are providing context, however it provides additional context ( or `initAnalyticsContext` )